src/main/java/org/spongepowered/common/mixin/core/block/MixinBlockPistonBase.java

```diff
@@ -88,4 +103,35 @@ private ImmutableDirectionalData getDirectionalData(IBlockState blockState) {
         return ImmutableDataCachingUtil.getManipulator(ImmutableSpongeDirectionalData.class,
                 DirectionResolver.getFor(blockState.getValue(BlockPistonBase.FACING)));
     }
+
+    @Inject(method = "doMove", at = @At("HEAD"))
+    public void onDoMoveHead(World worldIn, BlockPos pos, EnumFacing direction, boolean extending, CallbackInfoReturnable<Boolean> cir) {
```

Comment:
I personally prefer methods named more descriptively than `onDoMoveHead`. `startTrackingPhase` and `completeTrackingPhase` would be great.